app/views/events/index.html.erb

```diff
@@ -2,32 +2,33 @@
   <% Time.zone = @time_zone %>
   <h3 class="calendar-date-year"><%= Date::MONTHNAMES[@date.month]%> <%= @date.year %> </h3>
   <div class="container">
-    <%= button_to "#{Date::MONTHNAMES[Date.today.last_month.month]} #{Date.today.beginning_of_month.last_month.year}",
-                  calendar_path,
-                  class: 'z-button topic-button float-left',
-                  method: :get,
-                  params: {date: @date.beginning_of_month.last_month} %>
+    <%= button_to "#{Date::MONTHNAMES[@date.last_month.month]} #{@date.beginning_of_month.last_month.year}",
+
+      calendar_path,
+      class: 'z-button topic-button float-left',
+      method: :get,
+      params: {date: @date.beginning_of_month.last_month} %>
 
     <%= button_to "#{Date::MONTHNAMES[@date.next_month.month]} #{@date.beginning_of_month.next_month.year}",
-                  calendar_path,
-                  class: 'float-right z-button topic-button',
-                  method: :get,
-                  params: { date: @date.beginning_of_month.next_month } %>
+      calendar_path,
+      class: 'float-right z-button topic-button',
+      method: :get,
+      params: { date: @date.beginning_of_month.next_month } %>
 
-  </div>
-  <br/>
-  <br/>
-  <br/>
-  <div class="clearfix"></div>
-  <div class="row">
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Mon</div>
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Tues</div> 
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Wed</div>
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Thurs</div>
-  </div>
-  <br/>
-  <div class="container">
-    <% @events.keys.each do |week_num| %>
+</div>
+<br/>
+<br/>
+<br/>
+<div class="clearfix"></div>
+<div class="row">
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Mon</div>
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Tues</div> 
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Wed</div>
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Thurs</div>
+</div>
+<br/>
+<div class="container">
+  <% @events.keys.each do |week_num| %>
     <div class="row">
       <% if @events.keys.min == week_num %>
         <% (4 - @events[week_num].keys.length).times do %>
```

Comment:
Why `4`? What's `4`?

Reply:
4 is the week length for the calendar.  This conditional is how the padding on the first week of the month is displayed.  It creates offset columns by calculating the difference between the days in the week subtracted from the days per week displayed.  

I have created an instance variable in the controller to abstract the magic number.